fish_prompt.fish

```diff
@@ -81,9 +81,8 @@ function fish_prompt
     end
 
     # Check if there is an upstream configured
-    set -l has_upstream (command git rev-parse --abbrev-ref @'{u}' ^/dev/null)
-
-    if test -n "$has_upstream"
+    command git rev-parse --abbrev-ref '@{upstream}' >/dev/null ^&1
+    if test $status -eq 0
```

Comment:
I agree that it would be much nicer to have a more explicit check. See the comment above for why it wouldn't work previously.
Now I realized I forgot the Fish way to do these sorts of tests. Will submit a better version in a moment.